### PR TITLE
Add GitHub README with documentation

### PR DESCRIPTION
**Pull request by baijum**
Refer: https://issues.redhat.com/browse/HELM-159

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,346 @@
-#  Chart repo
+# OpenShift Helm Charts Repository
+
+OpenShift Helm Charts Repository is the canonical source of all the certified
```
Comment (Srivaralakshmi) [inline README.md:3]: canonical source to single source of truth

File: README.md [hunk 2]
```diff
@@ -1 +1,346 @@
...
+```
+
+The name of the chart that you are submitting must match the value in the OWNERS
+file.  The following section explains how to submit a chart source.  And the
+section followed describes how to submit a chart verification report—finally
+submitting a chart and the report together.
+
+### Chart Source
+
+The first step is to create a directory with the version number as the name. For
```
Comment (Srivaralakshmi) [inline README.md:88]: If this procedure has only one step, then just bullet list the step.

File: README.md [hunk 3]
```diff
@@ -1 +1,346 @@
...
+example, if you are publishing the &quot;0.1.0&quot; version of the
+&quot;awesome&quot; chart, create a directory like this:
+
+```
+charts/partners/acme/awesome/0.1.0/
+```
+
+The chart could be a tarball created using the `helm package` command or a
+directory with the chart source. If it's a tarball, it can be placed directly
```
Comment (Srivaralakshmi) [inline README.md:97]: it's: it is

File: README.md [hunk 4]
```diff
@@ -1 +1,346 @@
...
+    │   ├── NOTES.txt
+    │   ├── serviceaccount.yaml
+    │   ├── service.yaml
+    │   └── tests
+    │       └── test-connection.yaml
+    ├── values.schema.json
+    └── values.yaml
+```
+
+### Chart Verification Report
```
Comment (Srivaralakshmi) [inline README.md:134]: Submitting a chart verification report without the chart

File: README.md [hunk 5]
```diff
@@ -1 +1,346 @@
...
+
+When submitting a chart verification report without the chart, create a
```
Comment (Srivaralakshmi) [inline README.md:136]: If possible, it is nice to put the steps in an ordered list.

File: README.md [hunk 6]
```diff
@@ -1 +1,346 @@
...
+this:
+
+```
+charts/partners/acme/awesome/0.1.0/
+```
+
+Generate the report and save it under `0.1.0` with a given file name as
+`report.yaml`.
+
+### Chart with Verification Report
```
Comment (Srivaralakshmi) [inline README.md:148]: Submitting a chart verification report with the chart

File: README.md [hunk 7]
```diff
@@ -1 +1,346 @@
...
+
+You can also submit a chart and the report together. As mentioned in the "Chart
```
Comment (Srivaralakshmi) [inline README.md:150]: If possible, it is nice to put the steps in an ordered list.

File: README.md [hunk 8]
```diff
@@ -1 +1,346 @@
...
+Source" section, place the source or tarball under the version numbered
+directory. Similarly, as mentioned in the "Chart Verification Report" section,
+place `report.yaml` also under the save under the version numbered
+directory.
+
+## Troubleshooting Pull Request Failures
+
+After submitting the pull request, you can keep the pull request page open to
+observe the results.  If there is no issue with the pull request, it's going go
```
Comment (Srivaralakshmi) [inline README.md:159]: it's: it is

File: README.md [hunk 9]
```diff
@@ -1 +1,346 @@
...
+be merged automatically.  If there is any issue with with PR, a comment with
+details about the issue will appear.
+
+### Error when submitting files not part of any chart
+
+If you try to submit changes not related to any particular chart, you will
```
Comment (Srivaralakshmi) [inline README.md:165]: If you try to submit changes not related: If you try to submit changes that are not related

File: README.md [hunk 10]
```diff
@@ -1 +1,346 @@
...
+receive an error like this:
+
+```
+[ERROR] The files modified are not part of any chart
+```
+
+Please ensure you are making changes inside the chart version directory.  For
```
Comment (Srivaralakshmi) [inline README.md:172]: Is this the suggested action for the user to mitigate the above error? or is it the resolution?

File: README.md [hunk 11]
```diff
@@ -1 +1,346 @@
...
+
+If the directory structure (organization name) is wrong, please reach out to
+[Technology Partner Success Desk][partner-success-desk].  If you are a community
+member, you can send a separate pull request with a renaming of organization
+change.
+
+### Chart name mismatch with the directory structure
+
+The directory structure and the value of chart name in the `OWNERS` file must
+match.  Otheriwise you will see an error like this:
```
Comment (Srivaralakshmi) [inline README.md:225]: Otheriwise: Otherwise

Or you can use: Else,

File: README.md [hunk 12]
```diff
@@ -1 +1,346 @@
...
+
+If the directory structure (chart name) is wrong, please reach out to
+[Technology Partner Success Desk][partner-success-desk].  If you are a community
+member, you can send a separate pull request with a renaming of chart name
+change.
+
+### Error when submitting both chart source and tarball
+
+When submitting a chart, it should be either a tarball or the chart source
+source dirctory.  Otheriwise, you will see an error like this:
```
Comment (Srivaralakshmi) [inline README.md:244]: Otheriwise: Otherwise

Or you can use: Else,

Comment (Srivaralakshmi) [inline README.md:244]: chart source source dirctory: chart source directory

File: README.md [hunk 13]
```diff
@@ -1 +1,346 @@
...
+
+```
+[ERROR] Error analysing the report: ...
+```
+
+If the error is still persisting after upgrading to latest chart-verifier,
+please contact [Technology Partner Success Desk][partner-success-desk].  If you
+are community member, please open a [new issue][new-issue].
+
+If the reprt has some failure, it will be displayed like this:
```
Comment (Srivaralakshmi) [inline README.md:297]: reprt: report

File: README.md [hunk 14]
```diff
@@ -1 +1,346 @@
...
+3. Create a branch and make your required changes and send a pull request to
+   your `main` branch.
+
+### Can I use any command-line interface to create pull request?
+
+Yes, you can use the [GitHub CLI to create pull request][gh-cli-pr].
+
+### How to update OWNERS file?
+
+Partners can refer the [partner documentation][partners]. Community members can
```
Comment (Srivaralakshmi) [inline README.md:324]: can refer the: can refer to the

File: README.md [hunk 15]
```diff
@@ -1 +1,346 @@
...
+create a pull request with OWNERS file change.  The OWNERS file change should be
+a separate pull request and there should not be any chart related changes in the
+same PR.
+
+## Support
+
+You can use the issue tracker in this repository to report bugs.  If you are a
+partner, please refer [Technology Partner Success Desk
```
Comment (Srivaralakshmi) [inline README.md:332]: please refer: please refer to the

**Review by Srivaralakshmi — COMMENTED** [review]
I have put in my review comments. PTAL. Thanks!

File: README.md (hunk 1 above)
Comment (baijum) [reply README.md:3]: I used "canonical source" based on this README:
https://github.com/operator-framework/community-operators#about-this-repository
"single source of truth" is too technical and not really required in the first line of the document.

File: README.md [hunk 16]
```diff
@@ -1 +1,346 @@
-#  Chart repo
+# OpenShift Helm Charts Repository
+
+OpenShift Helm Charts Repository is the canonical source of all the certified
+Helm charts from [partners][partners] and the community. Once your chart is
+ready for submission, you can submit it as a pull request. The chart is going to
+be certified after running the [chart-verifier][chart-verifier] tool against
+it. You may also consider running the chart-verifier locally and submit the
+generated report for certification.
```
Comment (mmulholla) [inline README.md:8]: Struggling to word this well but need to be clear in the full range of options:
 
Once your chart is ready for for submission, you can submit a pull request. The pull request can include the chart and/or a report generated by the [chart-verifier][chart-verifier] tool. Based on submission the content the chart will be certified and the chart-verifier will be run if a report is not provided.

File: README.md (hunk 2 above)
Comment (mmulholla) [inline README.md:88]: This step is also required if you submit a report only. need to separate it form the submitting a chart procedure.

File: README.md [hunk 17]
```diff
@@ -1 +1,346 @@
...
+   * [Frequently Asked Questions](#frequently-asked-questions)
+      * [Can I test the pull request in my fork before submitting?](#can-i-test-the-pull-request-in-my-fork-before-submitting)
+      * [Can I use any command-line interface to create pull request?](#can-i-use-any-command-line-interface-to-create-pull-request)
+      * [How to update OWNERS file?](#how-to-update-owners-file)
+   * [Support](#support)
+<!--te-->
+
+---
+
+## Submitting a Chart
```
Comment (mmulholla) [inline README.md:50]: Would like to see here a summary of the submission options available:
- Just a report
- Chart tarball with or without report
- Chart src with or without report.

They need to be given equal weight.

**Comment by mmulholla** [comment]
Would like to see a summary of the steps used to certify the pull request. For example
- Check PR includes only chart.
- Check chart is in correct directory.
- If report is provided, confirm all mandatory checks are present and have passed.
    -   if chart is also provided check sha in report match sha calculate for submitted chart
- If report is not provided, run the report, confirm all mandatory checks pass. 


Also we need to provided some information on an exception process, so partner would like pull request merged despite failures.

File: README.md (hunk 10 above)
Comment (baijum) [reply README.md:172]: resolution

**Comment by baijum** [comment]
@mmulholla @Srivaralakshmi I have addressed all the review comments.

@jasperchui I have removed community-related sentences.

**Comment by baijum** [comment]
> Also we need to provided some information on an exception process, so partner would like pull request merged despite failures.

@mmulholla I am not sure about any process for exception (exemption?). Are we going to allow PR to be merged even if some checks fail? Who needs to approve such kinds of pull requests?

**Comment by baijum** [comment]
> Would like to see a summary of the steps used to certify the pull request.

Done. 18e291ba3bb9cb6ba0e006b20d2449b76032b74c

**Review by dperaza4dustbit — APPROVED** [review]
